crates/matrix-sdk/src/widget/client/implementation/openid.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use ruma::api::client::account::request_openid_token::v3::Response as RumaOpenIdResponse;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct OpenIdState {
```

Comment:
This isn't supposed to really be public, right?

```suggestion
pub(crate) struct OpenIdState {
```

Reply:
Also, why do all the Rust fields have different names from the ones used in JSON?

Reply:
Updated.